src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -0,0 +1,468 @@
+                .ext(prebidJsonNodes)
+                .banner(setUpBanner())
+                .build();
+        final AuctionContext auctionContext = setupAuctionContext(imp);
+        event = AuctionEvent.builder()
+                .auctionContext(auctionContext)
+                .bidResponse(auctionContext.getBidResponse())
+                .build();
+
+        target = new GreenbidsAnalyticsReporter(
+                greenbidsAnalyticsProperties,
+                jacksonMapper,
+                httpClient,
+                clock,
+                prebidVersionProvider);
+
+        // when
+        final Future<Void> result = target.processEvent(event);
+
+        //then
+        assertTrue(result.failed());
+        assertThat(result.cause())
+                .hasMessageStartingWith("Error decoding imp.ext.prebid: "
+                        + "Cannot construct instance of `org.prebid.server.proto.openrtb.ext.request.ExtOptions`");
+    }
+
+    private static <K, V> Map<K, V> doubleMap(K key1, V value1, K key2, V value2) {
+        final Map<K, V> map = new HashMap<>();
+        map.put(key1, value1);
+        map.put(key2, value2);
+        return map;
+    }
+
+    private static AuctionContext setupAuctionContext(Imp imp) {
+        // bid request
+        final Site site = Site.builder()
+                .domain("www.leparisien.fr")
+                .build();
+
+        final BidRequest bidRequest = BidRequest.builder()
+                .id("request1")
+                .imp(Collections.singletonList(imp))
+                .site(site)
+                .ext(setUpExtRequest())
+                .build();
+
+        // bid response
+        final Bid bid = Bid.builder()
+                .id("bid1")
+                .impid("imp1")
+                .price(BigDecimal.valueOf(1.5))
+                .adm("<div>Ad Markup</div>")
+                .build();
+
+        final SeatBid seatBidWithBid = SeatBid.builder()
+                .bid(Collections.singletonList(bid))
+                .seat("seat1")
+                .build();
+
+        final BidResponse bidResponse = BidResponse.builder()
+                .id("response1")
+                .seatbid(Collections.singletonList(seatBidWithBid))
+                .cur("USD")
+                .build();
+
+        final BidRejectionTracker bidRejectionTracker = new BidRejectionTracker(
+                "seat2",
+                Set.of("imp1"),
+                1.0);
+
+        bidRejectionTracker.reject("imp1", BidRejectionReason.NO_BID);
+
+        return AuctionContext.builder()
+                .httpRequest(HttpRequestContext.builder().build())
+                .bidRequest(bidRequest)
+                .bidResponse(bidResponse)
+                .bidRejectionTrackers(
+                        singletonMap(
+                                "seat2",
+                                bidRejectionTracker))
+                .build();
+    }
+
+    private static Banner setUpBanner() {
```

Comment:
nitpick: I'd prefer instead of `setUp...` use `given...`, like `givenBanner`, `givenAuctionContext`